### PR TITLE
saga-4.295: Adding a new bottom variant of Modal for mobile screens

### PR DESCRIPTION
**Pull request by juligobbi-fluxit**
In Asana: https://app.asana.com/0/1200871770472728/1200891873931394/f

Creating a new bottom position variant of Modal for mobile screens.

- Adding position props: 
    -centered (as default)
    -bottom
- New layout for scrolling, with a container for header and body modal
- New layout header inner which contains title, subtitle, and close button*

Note:
*It would be nice to refactor the default layout with this header implementation.

 
https://asanal.ink/21/295

### CODE REVIEW

File: src/components/Dialog/_private/dialogContent.tsx [hunk 1]
```diff
@@ -35,43 +40,101 @@ const DialogContent = React.forwardRef<HTMLInputElement, DialogContentProps>(
...
-                <Card
-                    customCss={cardStyles}
-                    backgroundColor={backgroundColor}
-                >
-                    {showCloseButton && (
-                        <div
-                            data-testid="close-button"
-                            css={closeButton}
-                            onClick={handleCloseButtonClick}
+            <Fragment>
```
Comment (BrandonDenham) [inline src/components/Dialog/_private/dialogContent.tsx:43]: most of this code is duplicated in either position.  This shouldn't be done this way, it looks like position barely affects the difference in either implementation.  don't know what we need the position == centered or position == bottom

also, we always use strict equality ===

**Review by BrandonDenham — CHANGES_REQUESTED** [review]

File: src/components/Dialog/_private/dialogContent.tsx (hunk 1 above)
Comment (juligobbi-fluxit) [reply src/components/Dialog/_private/dialogContent.tsx:43]: I was thinking..Maybe I could do 2 different things to solve this:

1. position centered and bottom with just an if on the main container. I need different styles for both of them at this high level, so I need to differentiate them

2. make a refactoring for Dialog, implementing a wrapper for header and a wrapper for body (as I did on the bottom layout), and even I could add a wrapper for footer (to separate the buttons)

Comment (rmcarias) [reply src/components/Dialog/_private/dialogContent.tsx:43]: BTW, good tip to know shortcut for Fragment in React is
```
<></>
```

**Review by rmcarias — CHANGES_REQUESTED** [review]

**Comment by rmcarias** [comment]
@juligobbi-fluxit  is this ready for review again?
CC: @jmoyers14

File: src/components/Dialog/_private/dialogContent.types.ts [hunk 2]
```diff
@@ -8,5 +8,6 @@ export default interface DialogContentProps {
     header?: DialogHeader;
     backgroundColor: Colors;
     showCloseButton?: boolean;
+    position?: string;
```
Comment (rmcarias) [inline src/components/Dialog/_private/dialogContent.types.ts:11]: one thing you can do here to help TS infer the type better is constrain it
position?: 'centered' | 'bottom' | undefined

**Review by rmcarias — APPROVED** [review]
just a minor comment on the position prop

File: src/components/Dialog/_private/dialogContent.types.ts (hunk 2 above)
Comment (juligobbi-fluxit) [reply src/components/Dialog/_private/dialogContent.types.ts:11]: Great! I love those improvements

**Review by rmcarias — APPROVED** [review]

**Review by BrandonDenham — APPROVED** [review]

**Pull request merged**